script.js

```diff
@@ -347,12 +347,14 @@ function addUpStudentCredits(reportCardTableElement) {
 function calculateSemesterGpa(reportCardTableElement) {
   let totalPoints = 0.0;
   let gpaElems = reportCardTableElement.querySelectorAll('.gpa');
+  let classesCount = 0;
   for (let elem of gpaElems) {
     if (elem.innerHTML in gpaPointsLookup) {
+      classesCount += 1;
       totalPoints += gpaPointsLookup[elem.innerHTML];
     }
   }
```

Comment:
Another way to structure this is to use a combination of array functions like reduce and filter. For example:

```
const filteredGpaElems = Array.from(gpaElems).filter(el => el.innerHTML in gpaPointsLookup);
const totalPoints = filteredGpaElems.reduce(sumTotalPoints, 0);

const gpa = totalPoints / filteredGpaElems.length;
```

A little more about using `in`. Another way to check that an object has a property is to use [Object.prototype.hasOwnProperty()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty). Using `in` would return true if the property you are checking is in the prototype chain. This would mean that it could return true for strings are not in the `gpaPointsLookup` object. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in#inherited_properties